eng/ci/templates/scripts/setup-e2e-tests.ps1

```diff
@@ -0,0 +1,156 @@
+#Requires -Version 6
+
+param(
+    [Switch]
+    $SkipStorageEmulator,
+
+    [Switch]
+    $SkipCoreTools,
+
+    [String]
+    $RepoRoot
+)
+
+$FunctionsRuntimeVersion = 4
+
+if (!$IsWindows -and !$IsLinux -and !$IsMacOs)
+{
+    # For pre-PS6
+    Write-Host "Could not resolve OS. Assuming Windows."
+    $IsWindows = $true
+}
+
+if($SkipCoreTools)
+{
+  Write-Host
+  Write-Host "---Skipping Core Tools download---"  
+}
+else
+{
+  $arch = [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture.ToString().ToLowerInvariant()
+  if ($IsWindows) {
+      $os = "win"
+  }
+  else {
+      if ($IsMacOS) {
+          $os = "osx"
+      } else {
+          $os = "linux"
+      }
+  }
+
+  Write-Host ""
+  Write-Host "Using latest Core Tools release from GitHub..."
+
+  # GitHub API call for latest release
+  $releaseInfo = Invoke-RestMethod -Uri "https://api.github.com/repos/Azure/azure-functions-core-tools/releases/latest" -Headers @{ "User-Agent" = "PowerShell" }
+
+  $latestVersion = $releaseInfo.tag_name
+  Write-Host "`nLatest Core Tools version: $latestVersion"
+
+  # Look for zip file matching os and arch
+  $pattern = "Azure\.Functions\.Cli\.$os-$arch\..*\.zip$"
+  $asset = $releaseInfo.assets | Where-Object {
+      $_.name -match $pattern
+  }
+
+  if (-not $asset) {
+      Write-Error "Could not find a Core Tools .zip for OS '$os' and arch '$arch'"
+      exit 1
+  }
+
+  $coreToolsURL = $asset.browser_download_url
```

Comment:
Similar to a PR I just made, we should add a query string in case we see a caching issue again:

https://github.com/Azure/azure-functions-dotnet-worker/pull/3186/files